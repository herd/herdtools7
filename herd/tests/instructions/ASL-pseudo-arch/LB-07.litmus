ASL LB-pseudo-arch

{
  x = 0;
  y = 0;
  0: X1= x;
  0: X2= y;
}

constant one :: bits(64) = 1[63:0];

func T0(x::bits(64), y:: bits(64))
begin
  let read = read_memory(x, 64);
  let x0 = UInt(read);
  let data = one OR (read XOR read);
  write_memory(y, 64, data);
end

func f(read::bits(64)) => bits(64)
begin
  var t = read;
  t[0:] = '1';
  return t;
end

func T1(x::bits(64), y:: bits(64))
begin
  let read = read_memory(y, 64);
  let x1 = UInt(read);
  let data = f(read);
  write_memory(x, 64, data);
end

func main() => integer
begin
  let x = read_register(1);
  let y = read_register(2);

  T0(x, y);
  T1(x, y);

  return 0;
end

exists (0: T0.0.x0 = 1 /\ 0: T1.0.x1 = 1)
