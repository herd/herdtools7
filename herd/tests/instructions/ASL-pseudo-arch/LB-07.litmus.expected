Test LB-pseudo-arch Allowed
States 3
0:T0.0.x0=0; 0:T1.0.x1=0;
0:T0.0.x0=0; 0:T1.0.x1=1;
0:T0.0.x0=1; 0:T1.0.x1=0;
No
Witnesses
Positive: 0 Negative: 3
Condition exists (0:T0.0.x0=1 /\ 0:T1.0.x1=1)
Observation LB-pseudo-arch Never 0 3
Hash=d5c470ab099f6cf334821d6710a3e00f

